backend: auth: FuzzSanitizeClusterName times out in CI due to unbounded input size

## Describe the bug
`FuzzSanitizeClusterName` in `backend/pkg/auth/cookies_fuzz_test.go` intermittently fails in CI with `context deadline exceeded` after 30s because the fuzzer generates unbounded string inputs with no size cap, causing some iterations to run long enough to exhaust the fuzz budget.

Expected: fuzz test completes within the 30s `-fuzztime` budget.
Actual: `--- FAIL: FuzzSanitizeClusterName (31.00s) context deadline exceeded`

## To Reproduce
1. Run `cd backend && go test -fuzz=FuzzSanitizeClusterName -fuzztime=30s ./pkg/auth/`
2. Observe `context deadline exceeded` after 31s.

## Environment
- Installation type: N/A (backend CI only)
- Headlamp Version: main branch

## Are you able to fix this issue?
Yes (I will propose a PR).

## Additional Context
The fix is to add a 256 character input size guard inside the fuzz function so pathological inputs are skipped and each iteration stays bounded. Cluster names are not expected to exceed 256 characters in practice. This is not a regression; the fuzz test has always had unbounded input.

--- a/backend/pkg/auth/cookies_fuzz_test.go
+++ b/backend/pkg/auth/cookies_fuzz_test.go
@@ -46,6 +46,10 @@ func FuzzSanitizeClusterName(f *testing.F) {
 	validCharsRegex := regexp.MustCompile(`^[a-zA-Z0-9\-_]*$`)
 
 	f.Fuzz(func(t *testing.T, input string) {
+		if len(input) > 256 {
+			t.Skip()
+		}
+
 		result := auth.SanitizeClusterName(input)
 
 		// Invariant 1: Result should never be longer than 50 characters

